MarkDown parser runtime error due to improperly formatted markdown

The fuzzing introduced in #51 has uncovered a bug in the MarkDown parser. Given the input:

```MarkDown
|
```

You will get the following error:

```
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/ericcornelissen/wordrow/internal/wordmaps.parseTableHeader(0xc000010330, 0x1, 0x1, 0x4e2525, 0x5db2fc)
	/media/win/workspace/wordrow/internal/wordmaps/markdown.go:40 +0x43c
github.com/ericcornelissen/wordrow/internal/wordmaps.parseTable(0xc000010330, 0x1, 0x1, 0xc000059dd0, 0x0, 0x0, 0xc000010330)
	/media/win/workspace/wordrow/internal/wordmaps/markdown.go:61 +0x5b
github.com/ericcornelissen/wordrow/internal/wordmaps.parseMarkDownFile(0xc000059e58, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x4674c6, 0x5eca9d99)
	/media/win/workspace/wordrow/internal/wordmaps/markdown.go:93 +0x1eb
github.com/ericcornelissen/wordrow/internal/wordmaps.FuzzMarkDown(0x7f74acc01000, 0x1, 0x1, 0x3)
	/media/win/workspace/wordrow/internal/wordmaps/markdown_fuzz.go:6 +0x86
go-fuzz-dep.Main(0xc000059f40, 0x2, 0x2)
	go-fuzz-dep/main.go:36 +0x1ad
main.main()
	github.com/ericcornelissen/wordrow/internal/wordmaps/go.fuzz.main/main.go:17 +0x5d
exit status 2
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -16,6 +16,7 @@ Versioning].
 ### Bug Fixes
 
 - Fix program crash due to empty CLI argument. ([#42])
+- Fix program crash because of an incomplete MarkDown file. ([#58])
 
 ## [0.3.0-beta] - 2020-05-03
 
@@ -76,3 +77,4 @@ Versioning].
 [#50]: https://github.com/ericcornelissen/wordrow/pull/50
 [#55]: https://github.com/ericcornelissen/wordrow/pull/55
 [#57]: https://github.com/ericcornelissen/wordrow/pull/57
+[#58]: https://github.com/ericcornelissen/wordrow/pull/58

--- a/internal/wordmaps/markdown.go
+++ b/internal/wordmaps/markdown.go
@@ -58,6 +58,10 @@ func parseTableHeader(tableLines []string) (rerr error) {
 // The error will be set if the table head or any table row has an incorrect
 // format.
 func parseTable(tableLines []string, wm *WordMap) (int, error) {
+	if len(tableLines) < 3 {
+		return 0, &parseError{"Incomplete table", tableLines[0]}
+	}
+
 	if err := parseTableHeader(tableLines); err != nil {
 		return 0, err
 	}

--- a/internal/wordmaps/markdown_test.go
+++ b/internal/wordmaps/markdown_test.go
@@ -239,12 +239,17 @@ func TestMarkDownIncorrectTableRow(t *testing.T) {
 	}
 }
 
-func TestMarkdownLineWithOnlyPipe(t *testing.T) {
-	markdown := `|`
+func TestMarkdownIncompleteTableEndOfFile(t *testing.T) {
+	markdown := `
+		# Foobar
+
+    | foo | bar |
+    | --- | --- |
+	`
 
 	_, err := parseMarkDownFile(&markdown)
 
 	if err == nil {
-		t.Fatal("Error should be set for line containing only a pipe ('|')")
+		t.Fatal("Error should be set for incomplete table at the end of the file")
 	}
 }

--- a/internal/wordmaps/markdown_test.go
+++ b/internal/wordmaps/markdown_test.go
@@ -243,8 +243,7 @@ func TestMarkdownIncompleteTableEndOfFile(t *testing.T) {
 	markdown := `
 		# Foobar
 
-    | foo | bar |
-    | --- | --- |
+		| foo | bar |
 	`
 
 	_, err := parseMarkDownFile(&markdown)

--- a/internal/wordmaps/markdown_test.go
+++ b/internal/wordmaps/markdown_test.go
@@ -238,3 +238,13 @@ func TestMarkDownIncorrectTableRow(t *testing.T) {
 		t.Errorf("Incorrect error message for (actual '%s')", err)
 	}
 }
+
+func TestMarkdownLineWithOnlyPipe(t *testing.T) {
+	markdown := `|`
+
+	_, err := parseMarkDownFile(&markdown)
+
+	if err == nil {
+		t.Fatal("Error should be set for line containing only a pipe ('|')")
+	}
+}

